tests/Unit/Evolution/Systems/Cce/Test_LinearSolve.cpp

```diff
@@ -0,0 +1,477 @@
+  using integration_modes_variables_tag =
+      ::Tags::Variables<db::wrap_tags_in<TestHelpers::RadialPolyCoefficientsFor,
+                                         integration_tags>>;
+
+  auto box = db::create<db::AddSimpleTags<
+      integration_variables_tag, Tags::BoundaryValue<BondiValueTag>,
+      integration_modes_variables_tag, Spectral::Swsh::Tags::LMax,
+      Spectral::Swsh::Tags::NumberOfRadialPoints>>(
+      db::item_type<integration_variables_tag>{number_of_grid_points},
+      db::item_type<Tags::BoundaryValue<BondiValueTag>>{
+          Spectral::Swsh::number_of_swsh_collocation_points(l_max)},
+      db::item_type<integration_modes_variables_tag>{
+          number_of_radial_polynomials},
+      l_max, number_of_radial_grid_points);
+
+  generate_powers_for_tags(make_not_null(&box), gen, make_not_null(&dist),
+                           number_of_radial_polynomials,
+                           tmpl::list<BondiValueTag>{});
+
+  // use the above powers to infer the powers for the derivative
+  db::mutate<
+      TestHelpers::RadialPolyCoefficientsFor<Tags::Integrand<BondiValueTag>>>(
+      make_not_null(&box),
+      [](const gsl::not_null<Scalar<ComplexModalVector>*> integrand_modes,
+         const Scalar<ComplexModalVector>& bondi_value_modes) noexcept {
+        for (size_t i = 0; i < get(bondi_value_modes).size() - 1; ++i) {
+          // sign change because these are modes of 1 - y.
+          get(*integrand_modes)[i] =
+              -static_cast<double>(i + 1) * get(bondi_value_modes)[i + 1];
+        }
+        get(*integrand_modes)[get(bondi_value_modes).size() - 1] = 0.0;
+      },
+      db::get<TestHelpers::RadialPolyCoefficientsFor<BondiValueTag>>(box));
+
+  const ComplexDataVector one_minus_y =
+      std::complex<double>(1.0, 0.0) *
+      (1.0 - Spectral::collocation_points<Spectral::Basis::Legendre,
+                                          Spectral::Quadrature::GaussLobatto>(
+                 number_of_radial_grid_points));
+
+  const auto random_angular_data = make_with_random_values<ComplexDataVector>(
+      gen, make_not_null(&dist),
+      Spectral::Swsh::number_of_swsh_collocation_points(l_max));
+
+  ComplexDataVector expected = outer_product(
+      random_angular_data,
+      radial_vector_from_power_series(
+          get(db::get<TestHelpers::RadialPolyCoefficientsFor<BondiValueTag>>(
+              box)),
+          one_minus_y));
+
+  generate_volume_data_from_separable(
+      make_not_null(&box), random_angular_data, one_minus_y,
+      tmpl::list<Tags::Integrand<BondiValueTag>>{});
+
+  make_boundary_data<BondiValueTag>(make_not_null(&box),
+                                    make_not_null(&expected), l_max);
+
+  db::mutate_apply<RadialIntegrateBondi<BondiValueTag>>(make_not_null(&box));
+
+  Approx numerical_differentiation_approximation =
+      Approx::custom()
+          .epsilon(std::numeric_limits<double>::epsilon() * 1.0e6)
```

Comment:
does this tolerance depend on the number of grid points? perhaps better pass a suitable tolerance to the `test_regular_integration` for each test?

Reply:
I think it's preferable to keep it in the test, because while it does depend on the number of grid points, it also depends on the order of the radial polynomial being used to create the function, which is explicitly set in the tests here.